### PR TITLE
[MB-8371] allow customer to remove secondary addresses

### PR DESCRIPTION
**Pull request by mboren**
### Bug description:
Expected:
When a service member selects "No" for secondary pickup address, delivery location, or secondary delivery location, the system should remove the address that "No" was selected for if we already had one saved previously.

Actual:
The system doesn't really do anything with the "No" selection, so the next time the page is loaded, the radio button is set to "Yes" and the address is still there.
Is there anything you would like reviewers to give additional scrutiny?

## Summary
There were multiple problems to fix:
- On the frontend, if an optional address had already been set, it would be sent in the update shipment payload regardless of whether the user selected "yes" or "no" for the optional address.
- When the backend gets a null address, it has no way to tell if an address had simply been omitted from a payload because it wasn't meant to be updated, or if it should be removed.

To address these problems, I added new API parameters: `HasSecondaryPickupAddress` and `HasSecondaryDeliveryAddress`. I did not address the primary delivery address.
If these parameters are set to `false` then the backend can know with confidence that we don't want the corresponding addresses anymore.
I also had to add these fields to the shipment model so that we could get them out of `payload_to_model` functions. That part felt kinda silly to me but I don't see a way around it.

Currently, it's just setting the address ids in the shipment to `null`, and orphaning the address records, instead of deleting them.
### Additional steps

- Log in as a customer and create an HHG shipment (if one does not already exist)
  - If you're using devseed data, I recommend logging in as `hhg@only.unsubmitted`
- Edit an HHG shipment.
- Add a secondary pickup address and a secondary delivery address
- save the shipment
- edit the shipment again, and remove the secondary pickup and delivery addresses
- Verify that they have actually been removed

https://user-images.githubusercontent.com/2627844/228091676-b32ff262-409e-4f6a-9b11-5ea32cbac8f8.mov

You should also be able to remove secondary addresses for NTS and NTS-R shipments.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database

### CODE REVIEW

**Comment by duncan-truss** [comment]
Removing the secondary addresses on the customer side worked for me! I don't know that this needs fixed but did observe 2 things:

1. On creation if secondary addresses are provided we don't set the has* fields in the request and coming back in the response those fields remain null.  It didn't seem to affect anything adversely though just a possible place for confusion.

2. I know we are limiting this to the secondary addresses, there is the case where the user can choose they don't know their main delivery address after initially providing one and it seems like this should remove the secondary address as well.  I think it's okay if we save this for the removing delivery address functionality work since you can still individually remove the secondary address does that sound right?

File: pkg/handlers/primeapi/payloads/payload_to_model.go [hunk 1]
```diff
@@ -283,13 +286,15 @@ func MTOShipmentModelFromUpdate(mtoShipment *primemessages.UpdateMTOShipment, mt
...
 		model.SecondaryPickupAddressID = &secondaryPickupAddressID
+		model.HasSecondaryPickupAddress = handlers.FmtBool(true)
 	}
 
 	addressModel = AddressModel(&mtoShipment.SecondaryDeliveryAddress.Address)
 	if addressModel != nil {
 		model.SecondaryDeliveryAddress = addressModel
 		secondaryDeliveryAddressID := uuid.FromStringOrNil(addressModel.ID.String())
 		model.SecondaryDeliveryAddressID = &secondaryDeliveryAddressID
+		model.HasSecondaryDeliveryAddress = handlers.FmtBool(true)
```
[the remark this answers is not in the document]
Comment (mboren) [reply pkg/handlers/primeapi/payloads/payload_to_model.go:297]: the prime API endpoint that uses this is a PATCH. And yeah, the secondary addresses aren't in the create shipment endpoint in the prime API at all. I wonder if that's functionality that we'll need to add.

File: swagger-def/internal.yaml [hunk 2]
```diff
@@ -2335,8 +2335,16 @@ definitions:
         $ref: 'definitions/Address.yaml'
       secondaryPickupAddress:
         $ref: 'definitions/Address.yaml'
+      hasSecondaryPickupAddress:
```
Comment (mboren) [inline swagger-def/internal.yaml:2338]: i will add descriptions to these new API fields

File: pkg/handlers/primeapi/payloads/payload_to_model.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (duncan-truss) [reply pkg/handlers/primeapi/payloads/payload_to_model.go:297]: Whoops thank you, I was looking at updateMTOShipmentAddress.

**Comment by mboren** [comment]
> 1. On creation if secondary addresses are provided we don't set the has* fields in the request and coming back in the response those fields remain null.  It didn't seem to affect anything adversely though just a possible place for confusion.

This has been fixed now, thanks!

> 2. I know we are limiting this to the secondary addresses, there is the case where the user can choose they don't know their main delivery address after initially providing one and it seems like this should remove the secondary address as well.  I think it's okay if we save this for the removing delivery address functionality work since you can still individually remove the secondary address does that sound right?

This sounds right to me.

**Review by duncan-truss — APPROVED** [review]

**Pull request merged**